### PR TITLE
Doc fix (crm_ends_with_ext comment) and enhancement (PE: source-dir-root structure)

### PR DESCRIPTION
**Pull request by jnpkrn**

### CODE REVIEW

**Comment by jnpkrn** [comment]
Without the PE part, https://wiki.clusterlabs.org/wiki/Bundle_Walk-Through
howto is still encumbered with some element of magic.

File: lib/common/strings.c [hunk 1]
```diff
@@ -293,8 +293,8 @@ crm_ends_with(const char *s, const char *match)
  *
  * \note Main incentive to prefer this function over \c crm_ends_with
  *       where possible is the efficiency (at the cost of added
- *       restriction on \p match as stated (the complexity class remains
- *       the same, though: BigO(M+N) vs. BigO(M+N+(M-N))).
+ *       restriction on \p match as stated; the complexity class
+ *       remains the same, though: BigO(M+N) vs. BigO(M+2N)).
```
Comment (kgaillot) [inline lib/common/strings.c]: one too many )

Comment (jnpkrn) [inline lib/common/strings.c]: I actually fixed the unbalancedness :)

Comment (kgaillot) [inline lib/common/strings.c]: ah, i see now :)

**Pull request merged**